tests/frontend/onnx/test_build_ir.py

```diff
@@ -59,3 +62,108 @@ def test_visit_value_info():
+
+    # update context
+    _update_context(graph, ctx)
+
+    # expected output before visinting the op node
+    expected_output_pre = """{'input1': <BlockArgument[tensor<0x0xf32>] index: 0, uses: 0>, 'input2': <BlockArgument[tensor<0x0xf32>] index: 1, uses: 0>}"""
+
+    assert str(ctx.value_by_name) == expected_output_pre
+
+    # visit node
+    visit_node(sub_node, ctx)
+
+    # expected output after visinting the op node
+    expected_output_post = """{'input1': <BlockArgument[tensor<0x0xf32>] index: 0, uses: 1>, 'input2': <BlockArgument[tensor<0x0xf32>] index: 1, uses: 1>, 'output': <OpResult[tensor<0x0xf32>] index: 0, operation: onnx.Add, uses: 0>}"""
+
+    assert str(ctx.value_by_name) == expected_output_post
+
+
+def test_visit_node_sub():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create graph composed only of one Sub operation
+    graph, sub_node = _create_graph_binary_op("Sub", "sub_graph")
+
+    # update context
+    _update_context(graph, ctx)
+
+    # expected output before visinting the op node
+    expected_output_pre = """{'input1': <BlockArgument[tensor<0x0xf32>] index: 0, uses: 0>, 'input2': <BlockArgument[tensor<0x0xf32>] index: 1, uses: 0>}"""
+
+    assert str(ctx.value_by_name) == expected_output_pre
+
+    # visit node
+    visit_node(sub_node, ctx)
+
+    # expected output after visinting the op node
+    expected_output_post = """{'input1': <BlockArgument[tensor<0x0xf32>] index: 0, uses: 1>, 'input2': <BlockArgument[tensor<0x0xf32>] index: 1, uses: 1>, 'output': <OpResult[tensor<0x0xf32>] index: 0, operation: onnx.Sub, uses: 0>}"""
+
+    assert str(ctx.value_by_name) == expected_output_post
+
+
+def _create_graph_binary_op(op_name: str, graph_name: str):
+    # define input and output names
+    input1_name = "input1"
+    input2_name = "input2"
+    output_name = "output"
+
+    # define op node
+    op_node = helper.make_node(
+        op_type=op_name,
+        inputs=[input1_name, input2_name],
+        outputs=[output_name],
+    )
+
+    # create graph (composed of just one operation)
+    graph = helper.make_graph(
+        nodes=[op_node],
+        name=graph_name,
+        inputs=[
+            helper.make_tensor_value_info(input1_name, TensorProto.FLOAT, [None, None]),
+            helper.make_tensor_value_info(input2_name, TensorProto.FLOAT, [None, None]),
+        ],
+        outputs=[
+            helper.make_tensor_value_info(output_name, TensorProto.FLOAT, [None, None]),
+        ],
+    )
+
+    return graph, op_node
+
+
+def _update_context(graph: GraphProto, ctx: OnnxXdslMapping):
+    name = graph.name
+    input_types = tuple(visit_value_info(input, ctx) for input in graph.input)
+    output_types = tuple(visit_value_info(output, ctx) for output in graph.output)
+
+    fn = func.FuncOp(name, (input_types, output_types))
+    with ImplicitBuilder(fn.body) as args:
+        for input, arg in zip(graph.input, args, strict=True):
+            ctx.value_by_name[input.name] = arg
```

Comment:
This feels like quite a complex helper to me. Could you please add a bit of code to do this manually? So something like this inlined into the tests above:
``` python
lhs = TestSSAValue()
rhs = TestSSAValue()
ctx = OnnxXdslMapping({
    "input1": lhs,
    "input2": rhs,
}, {})
op = visit_node(node, ctx)
assert op.lhs is lhs
assert op.rhs is rhs
...
```

Reply:
Ok, so instead of calling _update_context in the test, I should do this manually inside the test?

Reply:
Done

Reply:
Thank you